### PR TITLE
native: channel polish

### PR DESCRIPTION
**Pull request by dnbrwstr**
### Unreads/Scroller
- Mark channels read on load, regardless of unread banner position (temp, but I think it makes more sense to err on the side of marking read).
- Remove unread banners from notebooks + galleries
- Fix Android missing unread banner / scroll jump on load. (We’ll probably want to continue to tune this).
- Disable scroll-to-unread in notebooks and galleries
- Prevent unread banner from being modified while in channel

### Design
- Implement more intentional looking failure modes for gallery posts that fail to load.
- Refactor + polish `AuthorRow` and `GalleryPost`
- Fix reversed empty message on android: Fixes TLON-2244

### Misc
- Fix redbox error caused by undefined results from channel query
- Fix feature flag init warning
- Implement remote post ref fetching

### Known issues / Further work
- Visuals still need work, but should be closer to spec
- Refs sometimes fail to load on first attempt -- something wonky with the sub

### CODE REVIEW

File: apps/tlon-mobile/src/lib/featureFlags.ts [hunk 1]
```diff
@@ -25,7 +29,12 @@ export function isEnabled(name: FeatureName) {
 const storageKey = 'featureFlags';
 
 async function loadInitialState() {
-  const state = await storage.load({ key: storageKey });
+  let state: FeatureState | null = null;
+  try {
+    state = await storage.load({ key: storageKey });
+  } catch (e) {
+    // ignore
```
Comment (patosullivan) [inline apps/tlon-mobile/src/lib/featureFlags.ts:36]: Thank you

File: packages/shared/src/api/postsApi.ts [hunk 2]
```diff
@@ -89,6 +89,41 @@ export function chatAction(
...
+  channelId: string;
+  postId: string;
+  replyId?: string;
+}) {
+  const path = `/said/${channelId}/post/${postId}${
+    replyId ? '/' + replyId : ''
+  }`;
+  const data = await subscribeOnce<ub.Said>({ app: 'channels', path }, 3000);
+  const post = toPostReference(data);
+  // The returned post id can be different than the postId we requested?? But the
```
Comment (patosullivan) [inline packages/shared/src/api/postsApi.ts:106]: @Fang- @arthyn do we know what's up with this?

File: packages/shared/src/store/sync.ts [hunk 3]
```diff
@@ -18,7 +18,7 @@ import { addToChannelPosts, clearChannelPostsQueries } from './useChannelPosts';
 
 export { SyncPriority, syncQueue } from './syncQueue';
 
-const logger = createDevLogger('sync', false);
+const logger = createDevLogger('sync', true);
```
Comment (patosullivan) [inline packages/shared/src/store/sync.ts:21]:
```suggestion
const logger = createDevLogger('sync', false);
```

**Review by patosullivan — APPROVED** [review]
Looks great!

**Comment by jamesacklin** [comment]
Awesome. I'll build a one-off version for this and try it out before approving and merging.

**Comment by jamesacklin** [comment]
iOS built on-device feels good—@patosullivan were you able to build and run on the Android simulator?

**Comment by jamesacklin** [comment]
Fixes TLON-2244

**Comment by patosullivan** [comment]
Seeing consistent crashes on any scrolling in a chat channel on Android (same exception, stack trace we were seeing when scrolling back repeatedly in the scrollback):

```
 ERROR  Your app just crashed. See the error below.
java.lang.NullPointerException: Attempt to read from field 'int android.view.View.mViewFlags' on a null object reference in method 'void android.view.ViewGroup.dispatchGetDisplayList()'
  android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4538)
  android.view.View.updateDisplayListIfDirty(View.java:23660)
  android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4566)
  android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4539)
  android.view.View.updateDisplayListIfDirty(View.java:23660)
  android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4566)
  android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4539)
  android.view.View.updateDisplayListIfDirty(View.java:23660)
  android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4566)
  android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4539)
  android.view.View.updateDisplayListIfDirty(View.java:23660)
  android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4566)
  android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4539)
  android.view.View.updateDisplayListIfDirty(View.java:23660)
  android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4566)
  android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4539)
  android.view.View.updateDisplayListIfDirty(View.java:23660)
  android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4566)
  android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4539)
  android.view.View.updateDisplayListIfDirty(View.java:23660)
  android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4566)
  android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4539)
  android.view.View.updateDisplayListIfDirty(View.java:23660)
  android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4566)
  android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4539)
  android.view.View.updateDisplayListIfDirty(View.java:23660)
  android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4566)
  android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4539)
  android.view.View.updateDisplayListIfDirty(View.java:23660)
  android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4566)
  android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4539)
  android.view.View.updateDisplayListIfDirty(View.java:23660)
  android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4566)
  android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4539)
  android.view.View.updateDisplayListIfDirty(View.java:23660)
  android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4566)
  android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4539)
  android.view.View.updateDisplayListIfDirty(View.java:23660)
  android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4566)
  android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4539)
  android.view.View.updateDisplayListIfDirty(View.java:23660)
  android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4566)
  android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4539)
  android.view.View.updateDisplayListIfDirty(View.java:23660)
  android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4566)
  android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4539)
  android.view.View.updateDisplayListIfDirty(View.java:23660)
  android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:694)
  android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:700)
  android.view.ThreadedRenderer.draw(ThreadedRenderer.java:798)
  android.view.ViewRootImpl.draw(ViewRootImpl.java:5454)
  android.view.ViewRootImpl.performDraw(ViewRootImpl.java:5116)
  android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:4222)
  android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2836)
  android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:10145)
  android.view.Choreographer$CallbackRecord.run(Choreographer.java:1406)
  android.view.Choreographer$CallbackRecord.run(Choreographer.java:1415)
  android.view.Choreographer.doCallbacks(Choreographer.java:1015)
  android.view.Choreographer.doFrame(Choreographer.java:945)
  android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1389)
  android.os.Handler.handleCallback(Handler.java:959)
  android.os.Handler.dispatchMessage(Handler.java:100)
  android.os.Looper.loopOnce(Looper.java:232)
  android.os.Looper.loop(Looper.java:317)
  android.app.ActivityThread.main(ActivityThread.java:8674)
  java.lang.reflect.Method.invoke(Native Method)
  com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:580)
  com.android.internal.os.ZygoteInit.main(ZygoteInit.java:886)
```

**Review by patosullivan — CHANGES_REQUESTED** [review]
Causing frequent crashes on android

**Comment by dnbrwstr** [comment]
Request: since the crash is fixed, could someone resolve the conflict and merge in? There are a lotta changes in here, so would be good to do before things diverge more.

**Review by patosullivan — APPROVED** [review]

**Pull request merged**